base_layer/core/src/transactions/aggregated_body.rs

```diff
@@ -425,12 +431,17 @@ impl AggregateBody {
         &self,
         script_offset: PublicKey,
         factory: &CommitmentFactory,
+        prev_header: Option<HashOutput>,
+        height: Option<u64>,
     ) -> Result<(), TransactionError> {
         trace!(target: LOG_TARGET, "Checking script offset");
         // lets count up the input script public keys
         let mut input_keys = PublicKey::default();
+        let prev_hash: [u8; 32] = prev_header.unwrap_or_default().as_slice().try_into().unwrap_or([0; 32]);
+        let height = height.unwrap_or_default();
```

Comment:
I was going with trying to keep existing behavior. Currently, all scripts run with height = 0. The `Some(u64::MAX)` was only added and used in tests, not actual production code. 
This has a security implication, if we default to `u64::MAX` outside of tests, we might have scripts that should fail and pass. 
For example, if a wallet can't get a height for any reason, it should not assume that the tip is suddenly `u64::MAX` as this means that it will accept for example an HTLC refund transaction that should not be valid yet. This will leave observers to observe that the wallet is behaving badly.

Reply:
Alright, makes sense